assignments/this.js

```diff
@@ -1,26 +1,54 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. Without pointing to anything "this" with be bound to the window object.
+* 2. Implicit - when a function is called using dot notation the object before the . is what this points to 
+* 3. New- when a constructior function is used "this" points to the new object being created
+* 4. Explicit- When call or apply are used on a function 'this' is explictly bound
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
 
 // code example for Window Binding
+console.log(this)
 
 // Principle 2
 
 // code example for Implicit Binding
 
+let Me ={
+    "name":"Tom"
+}
+function likes(like1,like2,like3){
```

Comment:
`Me` here is actually an object. It is not a constructor function, but it can quickly be changed into one like so:
```
function Me () {
    this.name = "Tom";
}
```

Also, the `likes` function will not actually print any name because `this` isn't implicitly bound to the `Me` object. To fix this, you can add it into the `Me` constructor function definition like so (but you won't be able to call the method directly - you'll have to use `new` binding to do so):
```
function Me () {
    this.name = "Tom";
    this.likes = function (like1,like2,like3) {
        return `Hello, my name is ${this.name} and I like: ${like1}, ${like2}, ${like3}.`;
    }
}
```